misc/mdns/Cargo.toml

```diff
@@ -10,7 +10,7 @@ keywords = ["peer-to-peer", "libp2p", "networking"]
 categories = ["network-programming", "asynchronous"]
 
 [dependencies]
-async-std = "1.0"
+async-std = "1.4.0"
```

Comment:
I know you're a fan of bumping dependencies :smirk: but all the dependencies bumps are out of scope of this PR.

Reply:
This was actually a side-effect of temporarily using a fork of `async-std`.  Will fix.